### PR TITLE
pass rest of arguments to co to generator

### PR DESCRIPTION
**Pull request by bucaran**
Simply

```js
var args = slice.call(arguments, 1)
...
gen.apply(ctx, args)
```

Somewhat related https://github.com/tj/co/issues/156

### CODE REVIEW

**Comment by jonathanong** [comment]
thanks. can you add a test and rebase? there shouldn't be a "merge pull request" commit in a PR

**Comment by bucaran** [comment]
Gotcha.

Maybe I can bump, and add the changes to the CHANGELOG.md as well.

File: index.js [hunk 1]
```diff
@@ -42,12 +42,13 @@ co.wrap = function (fn) {
 
 function co(gen) {
   var ctx = this;
+  var args = slice.call(arguments, 1)
```
Comment (ruimarinho) [inline index.js]: What about inlining slicing using `new Array(arguments.length); (...)` in order to avoid v8 deoptimization?

Comment (bucaran) [inline index.js]: IMO that optimization should be done separately in another PR after this is merged.

**Comment by bucaran** [comment]
@jonathanong Ping.

**Comment by jonathanong** [comment]
should be a minor version bump because it's a new feature but i'll od that

**Comment by jonathanong** [comment]
added in 4.6.0. thanks! b54d18f8f472ad1314800e786993c4169a5ff9f8

**Comment by bucaran** [comment]
:+1: Thanks
